omod-1.9/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_9/ObsController1_9Test.java

```diff
@@ -258,6 +262,32 @@ public void getObs_shouldCreateAnObsWhenTheQuestionConceptIsSetAsAMapContainingT
 		assertEquals(++originalCount, getAllCount());
 	}
 	
+	@Test
+	public void shouldCreateObsAWithADateTimeValueHavingATimeZone() throws Exception {
+		String dateFormat = "yyyy-MM-dd'T'HH:mm:ss.SSSZ";
+		String dateWithTimeZone = "2018-11-13T00:00:06.040-0500";
+		executeDataSet("customConceptDataset1_9.xml");
+		
+		long originalCount = getAllCount();
+		SimpleObject obs = new SimpleObject();
+		obs.add("person", "da7f524f-27ce-4bb2-86d6-6d1d05312bd5");
+		SimpleObject concept = new SimpleObject();
+		concept.add("uuid", "A607c80f-1ea9-4da3-bb88-6276ce8868ii");
+		concept.add("name", new SimpleObject());
+		obs.add("concept", concept);
+		obs.add("obsDatetime", "2018-11-13T00:00:00.000-0500");
+		obs.add("value",dateWithTimeZone );
+		String json = new ObjectMapper().writeValueAsString(obs);
+		
+		Object obsWithTimeZone = deserialize(handle(newPostRequest(getURI(), json)));
+		assertEquals(++originalCount, getAllCount());
+		assertNotNull(PropertyUtils.getProperty(obsWithTimeZone, "value"));
+		
+		DateFormat sdf = new SimpleDateFormat(dateFormat);
+		Date date = sdf.parse(dateWithTimeZone);
+		assertEquals(sdf.format(date), PropertyUtils.getProperty(obsWithTimeZone, "value"));
```

Comment:
Doesn't something along this line 
`assertEquals(dateWithTimeZone, PropertyUtils.getProperty(obsWithTimeZone, "value"));` 
work fine than so that the test is as compact as possible?

Reply:
**assertEquals(dateWithTimeZone, PropertyUtils.getProperty(obsWithTimeZone, "value"));**
That would fail , because
 when i post a string `dateWithTimeZone`, it is first parsed and then converted. so it woudnt be exactly equal to the raw String `dateWithTimeZone`. 

so to get an equivalent of it , i still have to first do the conversion  as i have done 
             ```
DateFormat sdf = new SimpleDateFormat(dateFormat);
		Date date = sdf.parse(dateWithTimeZone);
```

Reply:
It looks like
 `dateWithTimeZone = "2018-11-13T00:00:06.040-0500";` is already of the format 
`dateFormat = "yyyy-MM-dd'T'HH:mm:ss.SSSZ";`

Reply:
thats true , but since this `2018-11-13T00:00:06.040-0500` is a plain String, when you convert it to Date object and then convert it back to String  , it slightly changes some how  . it may become some thing like`2018-11-13T00:00:06.03+0600  just an example. 
so thats y , wen u directly compare it like this here
`assertEquals(dateWithTimeZone, PropertyUtils.getProperty(obsWithTimeZone, "value"));`
it fails. you still first have to do the conversion as i did.